fineract-provider/src/main/java/org/apache/fineract/organisation/monetary/service/CurrencyReadPlatformServiceImpl.java

```diff
@@ -24,19 +24,15 @@
 import org.apache.fineract.infrastructure.core.domain.JdbcSupport;
 import org.apache.fineract.infrastructure.security.service.PlatformSecurityContext;
 import org.apache.fineract.organisation.monetary.data.CurrencyData;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.jdbc.core.RowMapper;
-import org.springframework.stereotype.Service;
 
-@Service
 public class CurrencyReadPlatformServiceImpl implements CurrencyReadPlatformService {
 
     private final JdbcTemplate jdbcTemplate;
     private final PlatformSecurityContext context;
     private final CurrencyMapper currencyRowMapper;
 
-    @Autowired
     public CurrencyReadPlatformServiceImpl(final PlatformSecurityContext context, final JdbcTemplate jdbcTemplate) {
```

Comment:
Please use @RequiredArgsConstructor.
The CurrencyMapper can also be written in one line (without a constructor):
```
private final CurrencyMapper currencyRowMapper = new CurrencyMapper();
```

Reply:
done!